src/NotesWriterSSC.cpp

```diff
@@ -439,16 +439,32 @@ static RString GetSSCNoteData( const Song &song, const Steps &in, bool bSavingCa
 			}
 		}
 		lines.push_back(ssprintf("#TECHCOUNTS:%s;", join(",", asTechCounts).c_str()));
-
-		std::vector<RString> asMeasureInfo;
-		FOREACH_PlayerNumber( pn )
+		
+		// NpsPerMeasure and NotesPerMeasure are stored differently from Radar Values and Tech Counts,
+		// because the number of measures is variable.
+		// For charts that have different steps per player, each set of values is separated
+		// with pipes "|".
+		// The vast majority of charts don't, so there's no reason to store duplicated data.
+		const std::vector<std::vector<float>> &allNpsPerMeasures = in.GetAllNpsPerMeasures();
+		std::vector<RString> npsPerMeasureStrings;
```

Comment:
You can preinitialize the size with either resize or constructor.
Will prevent any intermediate copies that may happen with push_back.
Same comments below.